Go/sereal/encode.go

```diff
@@ -330,6 +335,26 @@ func (e *Encoder) encodeDouble(by []byte, f float64) []byte {
 	return by
 }
 
+func (e *Encoder) encodeJsonNumber(by []byte, n json.Number, isKeyOrClass bool, strTable map[string]int) []byte {
+	if int64Value, err := n.Int64(); err == nil {
+		return e.encodeInt(by, reflect.Int, int64Value)
+	}
+
+	// we do not want to lose precision for large integers, as those are often IDs of things
+	// so if a value looks like a positive integer, but does not fit to int64, better keep it as a string
+	// uint64 values which don't fit to signed int are a common example
+	bigInt, isBigInt := new(big.Int).SetString(n.String(), 10)
```

Comment:
Maybe only try this branch if the previous error is `strconv.ErrRange`?

Or even better, if the previous error is `ErrRange`, just re-encode it straight to string?

Reply:
Yes, make sense to check for `strconv.ErrRange`. 

Also, on the second thought, stringifying large positive integers, but leaving large negative ones as floats with lost precision, goes against the principle of the least surprise, so yeah, we don't have to actually parse the number.

Updated the MR.